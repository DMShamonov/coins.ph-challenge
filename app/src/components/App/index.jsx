// @flow

import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import moment from 'moment';

import configureStore from 'redux/store';
import DevTools from 'components/DevTools';
import Router from 'components/Router';

moment.locale('en');

const store = configureStore();
const IS_DEVELOPMENT_MODE = process.env.NODE_ENV === 'development';

export default function App(): React.Node {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <Router />
        {IS_DEVELOPMENT_MODE && <DevTools />}
      </Provider>
    </BrowserRouter>
  );
}
