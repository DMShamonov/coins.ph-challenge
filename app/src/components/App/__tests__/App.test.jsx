import React from 'react';
import { shallow } from 'enzyme';

import App from '../index';

describe('`App` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
