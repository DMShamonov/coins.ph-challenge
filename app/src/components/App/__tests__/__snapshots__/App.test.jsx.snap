// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`App\` component should render correctly with default props 1`] = `
<BrowserRouter>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "liftedStore": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Router />
  </Provider>
</BrowserRouter>
`;
